include/zephyr/logging/log_ctrl.h

```diff
@@ -95,6 +95,17 @@ __syscall void log_panic(void);
  */
 __syscall bool log_process(void);
 
+/**
+ * @brief Process all pending log messages
+ */
+#if defined(CONFIG_LOG) && !defined(CONFIG_LOG_MODE_MINIMAL)
```

Comment:
flushing is needed only for deferred mode so that will be enough:
```suggestion
#ifdef CONFIG_LOG_MODE_DEFERRED
```

Reply:
log_flush can always be called this way even if the definition is empty which avoids leaving other code with ifdefs. I can add the defined(CONFIG_LOG_MODE_DEFERRED) in place of !defined(CONFIG_LOG_MODE_MINIMAL) if that is better

Reply:
Ok I realized what happened before and fixed it, though it added an ifdef in log_core.c around log_flush(void) this seems fine to me